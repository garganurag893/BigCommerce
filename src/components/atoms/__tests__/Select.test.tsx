import { render, fireEvent } from '@testing-library/react'

import Select from '../Select'

const options = ['BMW', 'Merceded']

describe('Test Select Component', () => {
  it('Check for Select tag in DOM', () => {
    const onClickHandler = jest.fn()
    const { getByTestId, getByText } = render(
      <Select options={options} className='test' onClick={onClickHandler} />,
    )

    expect(getByTestId('dropdown')).toBeInTheDocument()
    expect(getByText('BMW')).toBeInTheDocument()
    expect(getByTestId('dropdown')).toHaveClass('test')

    fireEvent.click(getByTestId('dropdown'))

    expect(onClickHandler).toHaveBeenCalledTimes(1)
  })

  it('Select Snapshot Test', () => {
    const { container } = render(<Select options={options} className='test' />)
    expect(container).toMatchSnapshot()
  })
})
