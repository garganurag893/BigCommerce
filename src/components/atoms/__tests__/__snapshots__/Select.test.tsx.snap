// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Select Component Select Snapshot Test 1`] = `
<div>
  <div
    class="relative"
  >
    <select
      class="test"
      data-testid="dropdown"
    >
      <option>
        BMW
      </option>
      <option>
        Merceded
      </option>
    </select>
    <div
      class="pointer-events-none absolute inset-y-0 right-0 flex items-center px-2 text-gray-700"
    >
      <svg
        class="fill-current h-4 w-4"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z"
        />
      </svg>
    </div>
  </div>
</div>
`;
